### PR TITLE
[23.2] Build param dict before creating entrypoint

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes https://github.com/galaxyproject/galaxy/issues/17438.

Thanks for the excellent debugging @cat-bro!

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

### CODE REVIEW

**Comment by sveinugu** [comment]
Right. So it seems there is a circular dependency afoot!

Moving `populate_interactivetools()` was crucial to allow injection of the interactive tool entry point path to the IT container, which again is crucial to get path-based interactive tools to work in most cases. So one needs to configure the interactive tool entry points before the entry point paths are to be injected to the job as e.g. an environment variable, such as in the following example:

```
<entry_points>
    <entry_point name="Display name" label="mytool" requires_domain="False" requires_path_in_url="True">
        <port>80</port>
    </entry_point>
</entry_points>

<environment_variables>
    <environment_variable name="EP_PATH" inject="entry_point_path_for_label">mytool</environment_variable>
</environment_variables>
```
(from https://docs.galaxyproject.org/en/master/admin/special_topics/interactivetools.html).

The only thing this breaks is the ability to use templates on the name of the entry points of IT which only seems to be used to put the name of the infile into the name of the entrypoint ("Display name" above). This again I believe is only really used for the "Active Interactive Tools" table (the title of the new window/tab is something else). So one alternative fix is to just remove this templating from all the ITs that use it, and not allow templating of entry point names. A more complicated fix is to refactor somehow to support both dependencies

**Comment by sveinugu** [comment]
Just to be clear: merging this PR would break path-based interactive tools!

**Comment by mvdbeek** [comment]
OK, I've moved the entry point creation so that the param_dict is available, but before templating out environment variables and config files.

**Comment by sveinugu** [comment]
LGTM. Thanks!

**Comment by mvdbeek** [comment]
Test failures are unrelated

File: test/unit/app/jobs/test_job_wrapper.py [hunk 1]
```diff
@@ -120,7 +120,7 @@ def set_compute_environment(self, *args, **kwds):
         pass
```
Comment (nsoranzo) [inline test/unit/app/jobs/test_job_wrapper.py:117]: The `populate_interactivetools` method at line 116-117 above can now be removed.

File: lib/galaxy/tools/evaluation.py [hunk 2]
```diff
@@ -841,4 +862,10 @@ def build(self):
         global_tool_logs(self._build_param_file, config_file, "Building Param File")
```
Comment (nsoranzo) [inline lib/galaxy/tools/evaluation.py:862]: Should we call `global_tool_logs(self._create_interactivetools_entry_points, config_file, "Building Interactive Tool Entry Points")` some lines above?
If not, maybe add a comment?

Comment (mvdbeek) [reply lib/galaxy/tools/evaluation.py:862]: I think the docstring in `RemoteToolEvaluator` covers that ?

**Review by nsoranzo — APPROVED** [review]

**Pull request merged**